packages/flutter/lib/src/widgets/localizations.dart

```diff
@@ -521,6 +530,7 @@ class _LocalizationsState extends State<Localizations> {
       key: _localizedResourcesScopeKey,
       locale: _locale,
       localizationsState: this,
+      loadGeneration: _loadGeneration,
```

Comment:
you could also just pass in `_typeToResources`, instead of tracking the generation number.